File: src/SkyBlock/enchants/armor/Molten.php
<?php


namespace SkyBlock\enchants\armor;


use pocketmine\player\Player;

class Molten extends BaseArmorEnchant {

    static int $id = 123;

    /**
     * @param Player $attacker
     * @param Player $victim
     * @param int    $enchantmentLevel
     */
    public function onActivation(Player $attacker, Player $victim, int $enchantmentLevel) : void {
        $this->sendActivation($victim, "§bMolten §aActivated!");
        $this->sendActivation($attacker, "§cStruck by §bMolten §cEnchant!");
        if (!$attacker->isOnFire()) {
            $attacker->setOnFire($enchantmentLevel);
        }
    }

}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/armor/Molten.java
package dev.skyblock.enchants.armor;

import dev.skyblock.enchants.BaseArmorEnchant;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import java.util.concurrent.ThreadLocalRandom;

public class Molten extends BaseArmorEnchant {
    public static final int ID = 123;

    public Molten() {
        super(ID, "Molten", 15);
    }

    @Override
    public boolean isApplicableTo(Player player, int level) {
        return ThreadLocalRandom.current().nextInt(1, 26) == 1;
    }

    @Override
    public void onActivation(Player victim, EntityDamageByEntityEvent ev, int level) {
        if (ev.getDamager() instanceof Player attacker) {
            victim.sendMessage("§bMolten §aActivated!");
            attacker.sendMessage("§cStruck by §bMolten §cEnchant!");
            attacker.setFireTicks(level * 20);
        }
    }
}
